editor/store/effects.js

```diff
@@ -71,26 +76,65 @@ export default {
 			content: getEditedPostContent( state ),
 			id: post.id,
 		};
+		const isAutosave = action.options && action.options.autosave;
+		let Model, newModel;
 
-		dispatch( {
-			type: 'UPDATE_POST',
-			edits: toSend,
-			optimist: { type: BEGIN, id: POST_UPDATE_TRANSACTION_ID },
-		} );
-		dispatch( removeNotice( SAVE_POST_NOTICE_ID ) );
-		const Model = wp.api.getPostTypeModel( getCurrentPostType( state ) );
-		new Model( toSend ).save().done( ( newPost ) => {
-			dispatch( {
-				type: 'RESET_POST',
-				post: newPost,
-			} );
+		if ( isAutosave ) {
```

Comment:
I know I told you to merge the two effects, but I'm starting to regret 🙃
I was originally thinking the `autosave` is just a save and should perform the same actions (update the post object etc...) but I was wrong as the autosave is something completely different now.
Seeing all these `if`s makes me think I was even more wrong to ask you to merge them.

Reply:
fine to separate them again; as you can see in the code, autosaving is a pretty different action than saving the post - for both drafts and published posts. 

To summarize the differences:
* at most one autosave is stored for each post for each user
* regular saves or updates (save draft or update publish post) create a revision each time (if something has changed)

Also:
* autosaves trigger a warning when the post editor is opened if newer and changed
* autosaves are marked as such is the revisions screen
* autosaves work with localstorage when a network connection is unavailable (not implemented yet)